api/config/v1alpha1/envoygateway_types.go

```diff
@@ -313,6 +320,35 @@ type ExtensionTLS struct {
 	CertificateRef gwapiv1b1.SecretObjectReference `json:"certificateRef"`
 }
 
+// EnvoyGatewayObservability defines the Envoy Gateway Observability configuration.
+type EnvoyGatewayObservability struct {
+
+	// Debug defines Debug Server configuration of Envoy Gateway
+	//
+	// +optional
+	Debug *DebugConfig `json:"debug,omitempty"`
+}
+
+// DebugConfig defines the Envoy Gateway Debug Server configuration.
+type DebugConfig struct {
+	// IPNetworkFamily defines the debug server IPNetworkFamily.
+	//
+	// +optional
+	// +kubebuilder:default=tcp
+	IPNetworkFamily string `json:"ipNetworkFamily,omitempty"`
+	// Port defines the port the debug server is exposed on.
+	//
+	// +optional
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:default=8899
+	Port int `json:"port,omitempty"`
+	// Host defines the debug server hostname.
+	//
+	// +optional
+	// +kubebuilder:default="0.0.0.0"
```

Comment:
Directly exposing golang pprof to 0.0.0.0 has security risks. Exposing it to 127.0.0.1 by default may be a better option.